test/calculator_test.js

```diff
@@ -0,0 +1,44 @@
+const assert = require('assert');
+const chai = require('chai');
+const calc = require('../calculator')
+
+describe('calculator', function () {
+  describe('#calculate()', function () {
+    it('adds one number', function () {
+      const calculator = calc.create_calculator('+');
+
+      calculator.add_number(16);
+
+      assert.equal(16, calculator.calculate());
+    });
+
+    it('adds two numbers', function () {
+      const calculator = calc.create_calculator('+');
+
+      calculator.add_number(5);
+      calculator.add_number(4);
+
+      assert.equal(9, calculator.calculate());
+    });
+
+    it('adds many numbers', function () {
+      const calculator = calc.create_calculator('+');
+
+      calculator.add_number(5);
+      calculator.add_number(4);
+      calculator.add_number(0);
+      calculator.add_number(-20);
+      calculator.add_number(1486);
+      calculator.add_number(-402);
+      calculator.add_number(4);
+
+      assert.equal(1077, calculator.calculate());
+    });
+
+    it('rejects the calculation when there are no numbers', function () {
+      const calculator = calc.create_calculator('+');
+
+      chai.expect(calculator.calculate).to.throw('Cannot calculate a result because no numbers were given');
+    });
+  });
```

Comment:
Could we maybe add a test for the other function add numbers.

Reply:
Wait.. you have.. could we maybe move them into another describe. They're all in the calculate describe, so it's making me think that they're all using the calculate function

Reply:
Instead could we add tests for the calculate function?

Reply:
... and all the +, -, *, and /

Reply:
Edit: I think I've understood this a bit wrong. So add_numbers just adds the numbers to an array which is then calculated. I was thinking add_numbers did addition whoops.. could we maybe change something to avoid the confusion. It still would be a good idea to write tests with -, * and /